src/agent.rs

```diff
@@ -481,16 +480,20 @@ impl AgentInner {
     /// Send a request with data appended
     async fn send_request_append_data(
         &self,
-        request: Request,
+        mut request: Request,
```

Comment:
We don't recommend to make parameter mutable, we can add a function to creating a new Request with new WorkRequestId based on the old one.